testgrid/cmd/summarizer/summarizer.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package summarizer converts input table data aggregated from test results
+// into the structured DashboardTabSummary proto for Testgrid consumption.
+package summarizer
+
+import (
+	"fmt"
+	"net/url"
+
+	"k8s.io/test-infra/testgrid/config"
+	"k8s.io/test-infra/testgrid/summary"
+)
+
+// TableToSummary converts input queryData collected from test results to
+// DashboardTabSummary proto.
+func TableToSummary(queryData map[string]interface{}, dashboardName string, dashboardTab config.DashboardTab, updateTimestamp int64) (summary.DashboardTabSummary, error) {
```

Comment:
Why is this a `map[string]interface{}` instead of a real type? What does it come from?

Reply:
Thanks Katharine. This comes from the Python implementation where the input is a Python dictionary with dynamic typing. Right now, I implemented it in such a way to be compatible with our existing Python input. I will sync up with @michelle192837 to see if we can nail down the exact type for OSS.

Reply:
New Response struct ported from internal. Reimplemented the TableToSummary() logic. Thanks for the feedback Katharine.